### ISSUE TITLE
Closing the running query window in Browser does not terminate the query (2.3.1)

### ISSUE THREAD

**Issue by genemishchenko** [neo4j/neo4j#6108]
I have some queries running indefinitely hogging the system.

When I close the window with the running query in the Browser, it briefly says "terminating" and closes the window, but from the reaource utilization I can see that the query execution still takes place and the  CPU time is still being consumed.

Only after I restart the instance the resource utilization goes back to normal.

Thanks.
Gene.

**Comment by jexp** [comment]
You mean the query frame, right?
Which browser?
Would you be able to share your database with me privately?

Any errors in graph.db/messages.log or data/log/console.log ?

**Comment by chrisvest** [comment]
From your #6104 issue, it sounds like it's the parsing, semantic checking or the planning of the query that takes a very long time. I don't think the transaction termination flag is checked in those phases; it's checked during the execution of queries. Perhaps it should also be checked in the earlier phases, but I think the root cause is the particular nature of the slow down you're seeing.

**Comment by genemishchenko** [comment]
Michael,

Yes, I meant the query frame.
I am using Chrome.
I shared the zipped version of my 2.3.1 database in the issue specifically related to performance tuning.
No, I don't see any error messages in messages.log. Sorry, I have trouble locating console.log.

Also, as I indicated in the other ticket, my 2.3.1 installation is really misbehaving now when I have 2.2.7 and 2.3.1 versions installed. And I dont even attempt to run them at the same time, I always make sure one is shut down.

That may be a separate issue.

Thanks.
Gene.

**Comment by jexp** [comment]
did you try `cypher planner=rule` as a workaround?

**Comment by genemishchenko** [comment]
Unfortunately, I did not get a chance to try reverting 2.3.1 to the rule-based optimizer before my installation started malfunctioning.

I think I will have to set up 2.3.1 on a separate machine, since it seems to be not working at all when installed on the same machine with 2.2.7.

**Comment by chrisvest** [comment]
By default, Neo4j expects to run on a dedicated machine, at least in terms of the default memory configurations. Make sure you configure the heap and the `dbms.pagecache.memory` settings such that there's enough memory to run the instances you want.

**Comment by genemishchenko** [comment]
I never ran more than one instance at a time, the other one was shut down completely.
When I run the 2.2.7 instance, it works fine.
When I run the 2.3.1 instance, it seems like it starts, but then I get the following errors in the Browser:

![neo4j_browser_err_1](https://cloud.githubusercontent.com/assets/9038701/11816847/50898e24-a320-11e5-9bc8-2fc1bcec8b98.PNG)
![neo4j_browser_err_2](https://cloud.githubusercontent.com/assets/9038701/11816853/52412164-a320-11e5-8ef9-1c541156915c.PNG)
![neo4j_browser_err_3](https://cloud.githubusercontent.com/assets/9038701/11816859/57be0e9a-a320-11e5-8d11-82bb6c5f9365.PNG)

**Comment by oskarhane** [comment]
This should not be the experience in more recent versions of neo4j-browser. Feel free to open an issue in https://github.com/neo4j/neo4j-browser if it still exists.
